fastlane/Fastfile

```diff
@@ -20,29 +23,46 @@ EXTRACTED_RELEASE_NOTES_PATH = File.join(RESOURCES_FOLDER, 'release_notes.txt')
 SENTRY_ORG_SLUG = 'a8c'
 SENTRY_PROJECT_SLUG = 'pocket-casts-ios'
 
+TEAM_ID_PRODUCTION = 'PZYM8XX95Q'
+
+CODE_SIGNING_STORAGE_OPTIONS = {
+  storage_mode: 's3',
+  s3_bucket: 'a8c-fastlane-match',
+  s3_region: 'us-east-2'
+}.freeze
+
+ASC_API_KEY_PRODUCTION = app_store_connect_api_key(
+  key_id: get_required_env("APP_STORE_CONNECT_API_#{TEAM_ID_PRODUCTION}_KEY_ID"),
+  issuer_id: get_required_env("APP_STORE_CONNECT_API_#{TEAM_ID_PRODUCTION}_ISSUER_ID"),
+  key_content: get_required_env("APP_STORE_CONNECT_API_#{TEAM_ID_PRODUCTION}_KEY_CONTENT")
+)
```

Comment:
Works 👌 

<img width="489" alt="image" src="https://github.com/Automattic/pocket-casts-ios/assets/1218433/a336ebda-8030-4468-8e19-d60376b2c089">

Reply:
Two notes:

 - One main drawback of having this as a global constant is that this will be evaluated no matter what lane is invoked—even if this is a lane that doesn't need any interaction with the ASC API, including running `fastlane lanes` to list available lanes–and would thus not only be not needed, but would also fail if one of the `get_required_env` vars is not defined locally 💥 
 - Is there any benefit of having our `.mobile-secrets/…/env` (and the `.env-default` file devs use locally) to use those compound env var names for those 3 values, and then having to pass the values explicitly… as opposed to directly use the ENV var names associated with the `ConfigItems` of `app_store_connect_api_key`—so that we don't have to pass any explicit parameter at all and this works out of the box?

---

My suggestion:

 - Update `~/.mobile-secrets/CI/secrets/pocket-casts-ios/env` to `export APP_STORE_CONNECT_API_KEY_KEY_ID="${APP_STORE_CONNECT_API_PZYM8XX95Q_KEY_ID}"` etc, so that those values (defined in `shared-secrets.yaml` are exposed as env vars matching the one from `fastlane action app_store_connect_api_key`.
 - In this `Fastfile`, remove the declaration of this global constant, and just use `api_key: app_store_connect_api_key` in the call sites instead of `api_key: ASC_API_KEY_PRODUCTION`. Given that all the parameters needed by `app_store_connect_api_key` would be provided by the env vars from `./mobile-secrets`, there is no need to provide them explicitly anymore, making its call site simpler.
 - Alternatively, instead of using `api_key: app_store_connect_api_key` in the parameters of the call sites, you can also just call `app_store_connect_api_key` first, and then the action you need without specifying any `api_key:` parameter, as those action propagate their values to one another via Lane Contexts (see https://docs.fastlane.tools/app-store-connect-api/#use-the-shared-value-in-lane-context)

Reply:
Addressed in [33b8162](https://github.com/Automattic/pocket-casts-ios/pull/1749/commits/33b81628e501c5e92b875608f0e205f22e2133f9)